src/partition.mjs

```diff
@@ -0,0 +1,42 @@
+import Dequeue from 'dequeue'
+import iterable from './iterable'
+
+const UNDONE = value => ({ done: false, value })
+const DONE = { done: true, value: undefined }
+
+function uncurried (func, iter) {
+  const satisfied = new Dequeue()
+  const unsatisfied = new Dequeue()
+  let exhausted = false
+
+  const iterator = iterable(iter)[Symbol.iterator]()
+  function add () {
+    const { value, done } = iterator.next()
+
+    if (done) {
+      exhausted = true
+    } else {
+      const chosen = func(value) ? satisfied : unsatisfied
+      chosen.push(value)
+    }
+  }
+
+  function part (queue) {
+    function next () {
+      if (queue.length) return UNDONE(queue.shift())
+      if (exhausted) return DONE
+      add()
+      return next()
```

Comment:
can you replace the recursive function with a simpler for loop ?

Reply:
Something like:
```js
while (true) {
     if (queue.length) return { done: false, value: queue.shift() }
      if (exhausted) return { done: true, value: undefined }
     add()
}
```

Reply:
I'm curious, what's wrong with recursive function? I find it more readable.

Reply:
First reason is just a matter of style. I found the while true better. But I understand this is subjective.

The second reason is about using the stack. The recursive version insert a lot of functions in the stack and this adds overhead (not big one but still ...). Also the stack have a limited amount of levels http://2ality.com/2014/04/call-stack-size.html

That is solved by tail call optimisation, but it is not widespread yet https://kangax.github.io/compat-table/es6/

So probably not a big deal, but given that the change is minimal ...

Reply:
I don't know, but here the change is super simple